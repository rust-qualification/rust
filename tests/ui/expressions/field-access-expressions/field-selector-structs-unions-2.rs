// Field Access Expressions: 6e237bb8-5e9d-4ea7-ad47-798f21044638

struct Point {
    x: i32,
    y: i32,
}

union MyUnion {
    x: i32,
    y: f64,
}

fn main() {
    let my_struct = Point { x: 1, y: 2 };
    let mut my_union = MyUnion { x: 5 };

    // invalid field access 
    let _ = my_union.1;
    //~^ ERROR no field `1` on type `MyUnion`
    let _ = my_struct.0; 
    //~^ ERROR no field `0` on type `Point`
    let _ = my_union.z;
    //~^ no field `z` on type `MyUnion`
    let _ = my_struct.z; 
    //~^ ERROR no field `z` on type `Point`
}
