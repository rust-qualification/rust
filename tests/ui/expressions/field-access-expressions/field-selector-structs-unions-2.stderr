error[E0609]: no field `1` on type `MyUnion`
  --> $DIR/field-selector-structs-unions-2.rs:18:22
   |
LL |     let _ = my_union.1;
   |                      ^ unknown field
   |
help: a field with a similar name exists
   |
LL |     let _ = my_union.x;
   |                      ~

error[E0609]: no field `0` on type `Point`
  --> $DIR/field-selector-structs-unions-2.rs:20:23
   |
LL |     let _ = my_struct.0; 
   |                       ^ unknown field
   |
help: a field with a similar name exists
   |
LL |     let _ = my_struct.x; 
   |                       ~

error[E0609]: no field `z` on type `MyUnion`
  --> $DIR/field-selector-structs-unions-2.rs:22:22
   |
LL |     let _ = my_union.z;
   |                      ^ unknown field
   |
help: a field with a similar name exists
   |
LL |     let _ = my_union.x;
   |                      ~

error[E0609]: no field `z` on type `Point`
  --> $DIR/field-selector-structs-unions-2.rs:24:23
   |
LL |     let _ = my_struct.z; 
   |                       ^ unknown field
   |
help: a field with a similar name exists
   |
LL |     let _ = my_struct.x; 
   |                       ~

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0609`.
